src/app/containers/App/App.jsx

```diff
@@ -1,95 +1,65 @@
-/* eslint-disable */
-/*
- * © Jordan Tart https://github.com/jtart
- * https://github.com/jtart/react-universal-app
- */
-import { Component } from 'react';
+import { useEffect, useState, useRef } from 'react';
 import { renderRoutes } from 'react-router-config';
 import { withRouter } from 'react-router-dom';
 import getRouteProps from '../../routes/getInitialData/utils/getRouteProps';
 
-import loadInitialData from '../../routes/loadInitialData';
+export const App = ({ routes, location, initialData, bbcOrigin }) => {
+  const { service, isAmp, id } = getRouteProps(routes, location.pathname);
 
-export class App extends Component {
-  constructor(props) {
-    super(props);
+  const [state, setState] = useState({
+    data: initialData,
+    service,
+    id,
+    isAmp,
+    loading: false,
+    error: null,
+  });
 
-    const { service, isAmp, id } = getRouteProps(
-      this.props.routes,
-      this.props.location.pathname,
-    );
+  const isInitialMount = useRef(true);
 
-    this.state = {
-      data: this.props.initialData,
-      service: service,
-      id: id,
-      isAmp: isAmp,
-      loading: false,
-      error: null,
-      loadInitialDataPromise: null,
-    };
-  }
+  useEffect(() => {
+    if (isInitialMount.current) {
+      isInitialMount.current = false;
+    } else {
+      // Only update on subsequent page renders
+      const {
+        service: newService,
```

Comment:
This is purely personal preference so feel free to ignore.

```suggestion
        service: nextService,
```